sessiontxn/txn.go

```diff
@@ -72,6 +74,28 @@ func CanReuseTxnWhenExplicitBegin(sctx sessionctx.Context) bool {
 	return txnCtx.History == nil && !txnCtx.IsStaleness && sessVars.SnapshotTS == 0
 }
 
+// CheckBeforeNewTxn is called before entering a new transaction. It checks whether the old
+// txn is valid in which case we should commit it first.
+func CheckBeforeNewTxn(ctx context.Context, sctx sessionctx.Context) error {
```

Comment:
`CheckBeforeNewTxn` is not a good name here. Maybe we can call it `DoCommitBeforeNewTxn()` ?

Reply:
Fine.